File: app/src/main/java/com/adisa/diningplus/network/entities/Item.java
package com.adisa.diningplus.network.entities;

import org.json.JSONException;
import org.json.JSONObject;

public class Item {
    public int id;
    public String name;
    public String ingredients;
    public String course;
    public boolean meat;
    public boolean animalProducts;
    public boolean alcohol;
    public boolean nuts;
    public boolean shellfish;
    public boolean peanuts;
    public boolean dairy;
    public boolean egg;
    public boolean pork;
    public boolean fish;
    public boolean soy;
    public boolean wheat;
    public boolean gluten;
    public boolean coconut;

    public boolean restricted;

    public static Item fromJSON(JSONObject raw) throws JSONException {
        Item item = new Item();
        item.id = raw.getInt("id");
        item.name = raw.getString("name");
        item.course = raw.getString("course");
        item.ingredients = raw.getString("ingredients");
        item.meat = raw.getBoolean("meat");
        item.animalProducts = raw.getBoolean("animal_products");
        item.alcohol = raw.getBoolean("alcohol");
        item.nuts = raw.getBoolean("nuts");
        item.shellfish = raw.getBoolean("shellfish");
        item.peanuts = raw.getBoolean("peanuts");
        item.dairy = raw.getBoolean("dairy");
        item.egg = raw.getBoolean("egg");
        item.pork = raw.getBoolean("pork");
        item.fish = raw.getBoolean("fish");
        item.soy = raw.getBoolean("soy");
        item.wheat = raw.getBoolean("wheat");
        item.gluten = raw.getBoolean("gluten");
        item.coconut = raw.getBoolean("coconut");
        return item;
    }
}

File: app/src/main/java/com/adisa/diningplus/network/entities/Item.kt
package com.adisa.diningplus.network.entities

import org.json.JSONObject

class Item(raw: JSONObject) {
    var id: Int = raw.getInt("id")
    var name: String = raw.getString("name")
    var ingredients: String = raw.getString("course")
    var course: String = raw.getString("ingredients")
    var meat: Boolean = raw.getBoolean("meat")
    var animalProducts: Boolean = raw.getBoolean("animal_products")
    var alcohol: Boolean = raw.getBoolean("alcohol")
    var treeNut: Boolean = raw.getBoolean("tree_nut")
    var shellfish: Boolean = raw.getBoolean("shellfish")
    var peanuts: Boolean = raw.getBoolean("peanuts")
    var dairy: Boolean = raw.getBoolean("dairy")
    var egg: Boolean = raw.getBoolean("egg")
    var pork: Boolean = raw.getBoolean("pork")
    var fish: Boolean = raw.getBoolean("fish")
    var soy: Boolean = raw.getBoolean("soy")
    var wheat: Boolean = raw.getBoolean("wheat")
    var gluten: Boolean = raw.getBoolean("gluten")
    var coconut: Boolean = raw.getBoolean("coconut")

    var restricted = false
}